New feature : Library

Fairspy opened the pull request:
## Description
- Added the library page showing the already rated books of the user
- Fixes #64 

## Affected Dependencies
No dependencies were affected

## How has this been tested?
All the previously, as well as newly defined tests, were passed 

## GIF Before
- 

![Before](https://user-images.githubusercontent.com/64915836/111191523-28e0ff00-85de-11eb-81e5-bcb3020ba563.gif)



## GIF After


![After(3)](https://user-images.githubusercontent.com/64915836/111193200-ed473480-85df-11eb-9471-e63419272478.gif)

![After(4)](https://user-images.githubusercontent.com/64915836/111193508-3bf4ce80-85e0-11eb-8cda-144df40a884b.gif)

(I couldn't add the whole after GIF together as the file size was too big)
## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/Praful932/Kitabe/blob/master/CONTRIBUTING.md).
- [X] My changes do not edit/add any unrequired files.
- [X] My changes are covered by tests.

Praful932 commented on BookRecSystem/settings.py:
```diff
@@ -24,6 +24,7 @@
 # SECURITY WARNING: keep the secret key used in production secret!
 SECRET_KEY = os.environ.get('KITABE_SECRET_KEY')
```
Unstage this file..

Praful932 commented on mainapp/helpers.py:
```diff
@@ -74,7 +74,7 @@ def get_rated_bookids(user_ratings):
     already_rated = []
     for rating in user_ratings:
-        book_id = rating.bookrating
+        book_id = rating.bookid
```
Nice catch, so then recommendation was a bit bugged earlier xD
@maneprajakta See this [Line](https://github.com/Praful932/Kitabe/blob/master/mainapp/views.py#L67)

maneprajakta replied:
Yep seems now better recommendation we will get :)

Praful932 commented on mainapp/tests.py:
```diff
@@ -209,3 +209,36 @@ def test_user_rated_book_valid(self):
+
+
+class RatedBooksTestCase(TestCase):
+    '''
+    Read books view test case
+    '''
```
```suggestion
    """Already Read Books View Test Case"""
```

Praful932 replied:
Would suggest to follow this for every function, we are currently updating our docs #36

Praful932 commented on mainapp/views.py:
```diff
@@ -79,3 +80,26 @@ def book_recommendations(request):
+
+@login_required
+@ensure_csrf_cookie
+def read_books(request):
```
Same here, follow numpy doc style for writing docstrings

Praful932 reviewed (changes requested):
I have added some comments, please resolve them

maneprajakta commented on static/mainapp/css/read.css:
```diff
@@ -0,0 +1,157 @@
+  }
+
+  .card-text {
+    font-family: "Pacifico", cursive;
```
font-family for card-text : "Courgette", cursive;
We are following this.

Praful932 replied:
@maneprajakta Is this pending?

maneprajakta replied:
@Praful932  Nope

maneprajakta reviewed (changes requested):
Requested Changes.

Fairspy commented:
@Praful932 @maneprajakta I've added the requested changes please review them.
Also, I tried to unstage settings.py but I'm not sure if it worked or not.

Praful932 reviewed (changes requested):
The file is still staged :)
Follow [this](https://stackoverflow.com/questions/38743912/remove-a-file-from-a-git-pull-request)

Fairspy commented:
@Praful932 I've tried to unstage settings.py let me know if it worked or not.

Praful932 commented:
@Fairspy Yep it's unstaged, There are some changes requested in the conversation, We haved added our reviewes, please resolve them

Fairspy commented:
@Praful932 I've added the requested changes please review them.
Let me know if any further changes are required.

Praful932 commented:
@Fairspy Please check the Files changed to resolve issues, there are still [some](https://github.com/Praful932/Kitabe/pull/100/files#r594573902) pending

Fairspy commented:
@Praful932 Is it okay?
I went through the numpy style guide but I am not completely sure about how to write docstrings for views if you need me to add something else please let me know.

Praful932 commented:
@Fairspy for now we have decided to keep docs on low prio, we'll make a new issue for that specifically
Please check [this](https://github.com/Praful932/Kitabe/pull/100#discussion_r594830970)

Praful932 commented on mainapp/urls.py:
```diff
@@ -5,6 +5,7 @@
     path('genre_books/<genre>', views.genre_books, name='genre_books'),
     path('explore_books/', views.explore_books, name='explore_books'),
     path('book_recommendations/', views.book_recommendations, name='book_recommendations'),
+    path('read_books/', views.read_books, name='read_books')
```
let's make this `library`

Fairspy commented:
@Praful932 I've changed the path to `library` and I had changed the font of `card-text` to `courgette` earlier.
Let me know if any further changes are required.

maneprajakta reviewed (changes requested):
- Fix Rate this button as it's on other pages.
![Screenshot from 2021-03-19 22-04-54](https://user-images.githubusercontent.com/57425995/111813557-59d37380-88ff-11eb-9b2b-fe4a8495f40c.png)
- Increase font size of book author

Praful932 commented:
@maneprajakta I'm thinking on the lines of how we should structure the library
- Library -> read_books
- Library -> to_read
then their path could be like this `/library/read_books` (This one)
& `/library/to_read`
Wdyt

maneprajakta commented:
> @maneprajakta I'm thinking on the lines of how we should structure the library
> 
> * Library -> read_books
> * Library -> to_read
>   then their path could be like this `/library/read_books` (This one)
>   & `/library/to_read`
>   Wdyt

Seems great to me :+1: 
In library page itself a left corner we provide both option and by default it would be read_books

Praful932 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/45713796/111869128-cacc6700-89a3-11eb-9594-6bde8a49397a.png)
Button width needs to be fixed

Fairspy commented:
@Praful932 I've fixed the button width you can check it out.
@maneprajakta I've increased the font-size of book author let me know if you want it changed further.
As for the behaviour of the rate book button I can't really see the same behaviour on my end so I don't really know how to go about fixing the issue are you seeing it for all the books or is it specifically for that one?

Fairspy commented:
I'm really sorry I force pushed to the branch and it deleted your previous commit I've committed the changes to urls.py again I'm really sorry.

Praful932 reviewed (approved):
Seems good to me now

Praful932 commented:
> I'm really sorry I force pushed to the branch and it deleted your previous commit I've committed the changes to urls.py again I'm really sorry.

Np, the rate button seems fixed now

Praful932 commented:
After revisting this @maneprajakta  Since this section is of already rated books, so instead of displaying `Rate this book` on every book, why don't we display the user's rating directly, so user does not confused

maneprajakta commented:
@Praful932 
yep a valid point.
I think we can create a new issue for this one.

Praful932 reviewed (approved):
Splendid Work @Fairspy This marks our 100th PR! 🚀
